core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -703,7 +720,67 @@ public void setDefaultTTL(int defaultTTL) {
                 .onErrorResumeNext(Observable.empty());
     }
 
+    /**
+     * Intended to be used at the startup of the MetricsServiceImpl to ensure we have enough tables for processing
+     */
+    public void verifyAndCreateTempTables() {
+        ZonedDateTime currentBlock = ZonedDateTime.ofInstant(Instant.ofEpochMilli(DateTimeService.now.get().getMillis()), UTC)
+                .with(DateTimeService.startOfPreviousEvenHour());
+
+        ZonedDateTime lastStartupBlock = currentBlock.plus(6, ChronoUnit.HOURS);
+        verifyAndCreateTempTables(currentBlock, lastStartupBlock).await();
+    }
+
+    @Override
+    public Completable verifyAndCreateTempTables(ZonedDateTime startTime, ZonedDateTime endTime) {
+        Set<Long> timestamps = new HashSet<>();
+
+        while(startTime.isBefore(endTime)) {
+            // Table sizes are not configurable at this point
+            timestamps.add(startTime.toInstant().toEpochMilli());
+            startTime = startTime.plus(2, ChronoUnit.HOURS);
+        }
+
+        return Completable.fromObservable(dataAccess.createTempTablesIfNotExists(timestamps));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Completable compressBlock(long startTimeSlice, int pageSize, int maxConcurrency) {
+        return Completable.fromObservable(
```

Comment:
There is a lot going on in this method both in terms of the work it is doing as well as the RxJava operators being used. It would be really helpful to do a BlueJeans session explain what is happen and the operators being used.